bundles/binding/org.openhab.binding.plex/src/main/java/org/openhab/binding/plex/internal/PlexConnector.java

```diff
@@ -0,0 +1,485 @@
+		logger.debug("Refreshing Plex sessions");
+		
+		MediaContainer container = getDocument(sessionsUrl, MediaContainer.class);
+		
+		if (container != null && container.getVideos() != null) {
+			Map<String, PlexSession> previousSessions = new HashMap<String, PlexSession>(sessions);
+			sessions.clear();
+			
+			for (Video video : container.getVideos()) {
+				Player player = video.getPlayer();
+				
+				PlexSession session = new PlexSession();
+				session.setSessionKey(video.getSessionKey());
+				session.setState(PlexPlayerState.of(player.getState()));
+				if (!StringUtils.isBlank(video.getGrandparentTitle())) 
+					session.setTitle(video.getGrandparentTitle() + " - " + video.getTitle());
+				else
+					session.setTitle(video.getTitle());
+				session.setType(video.getType());
+				session.setMachineIdentifier(player.getMachineIdentifier());
+				if (StringUtils.isNumeric(video.getDuration()))
+					session.setDuration(Integer.valueOf(video.getDuration()));
+				
+				setVolumeFromPreviousSession(previousSessions, session);
+				
+				sessions.put(video.getSessionKey(), session);				
+			}
+		}
+	}
+	
+	private void internalSendCommand(String machineIdentifier, String url) throws IOException {
+		logger.debug("Calling url {}", url);
+		
+		client.prepareGet(url)
+		.setHeader("X-Plex-Target-Client-Identifier", machineIdentifier)
+		.execute(new AsyncCompletionHandler<Response>() {
+			@Override
+			public Response onCompleted(Response response) throws Exception {
+				if (response.getStatusCode() != 200) {
+					logger.error("Error while sending command to Plex: {}\r\n{}", response.getStatusText(), response.getResponseBody());
+				}
+				return response;
+			}
+
+			@Override
+			public void onThrowable(Throwable t) {
+				logger.error("Error sending command to Plex", t);
+			}
+		});
+	}
+	
+	private PlexSession getSession(String sessionKey) {
+		if (!sessions.containsKey(sessionKey)) {
+			refreshSessions();
+		}
+		
+		if (sessions.containsKey(sessionKey)) 
+			return sessions.get(sessionKey);
+		
+		return null;
+	}
+	
+	private void setVolumeFromPreviousSession(Map<String, PlexSession> previousSessions, PlexSession newSession) {
+		String machineIdentifier = newSession.getMachineIdentifier();
+		for (Entry<String, PlexSession> session : previousSessions.entrySet()) {
+			if (session.getValue().getMachineIdentifier().equals(machineIdentifier)) {
+				newSession.setVolume(session.getValue().getVolume());
+				return;
+			}
+		}
+	}
+
+	private void connect() {
+		int delay = 0;
+		logger.debug("Reconnecting web socket to Plex");
+		while (!isConnected()) {
+			try {
+				Thread.sleep(delay);
+				open();
+			} catch (IOException e) {
+				logger.error("Error reconnecting to Plex", e);
+			} catch (InterruptedException e) {
+				logger.error("Interrupted while reconnecting to Plex", e);
+			} catch (ExecutionException e) {
+				logger.error("Error reconnecting to Plex", e);
+			}
+			delay = RECONNECT_DELAY;
```

Comment:
the method is called "connect()" but the logs speak from "reconnect()" - it would expect to log about "connecting"